### ISSUE TITLE
Enable opt-in analytics reporting for CLI usage

### ISSUE THREAD

**Issue by gnidan** [trufflesuite/truffle#1282]
# Overview

It's currently hard to answer the question of how many users Truffle has. In order to evaluate our usage over time, as well as to make informed decisions for future improvements, we should allow Truffle to collect anonymized usage metrics and publish them to Google Analytics.

# Requirements

- Must be opt-in.
- We should prompt the user when they first run Truffle, asking for their permission to enable metrics
- This preference should be saved for the user across all their Truffle projects, so likely this is dependent on #1281
- Should have some way to turn on and off easily. If we go with a `truffle config` command, I imagining something like:
  ```sh
  truffle config --global enable-analytics
  truffle config --global disable-analytics
  ```
- If analytics are enabled, metrics should be reported for command invocations and likely also whether or not those commands exited successfully or had errors.
- Metrics should get published to our Google Analytics account

**Comment by gnidan** [comment]
Not stale

**Comment by gnidan** [comment]
This is merged and released in beta
